content/java/concepts/array-list/terms/addall/addall.md

```diff
@@ -0,0 +1,81 @@
+---
+Title: '.addAll()'
+Description: 'Adds a collection to an ArrayList.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The `.addAll()` method is used to add the contents of a collection to an instance of the [`ArrayList`](https://www.codecademy.com/resources/docs/java/array-list) class.
+
+## Syntax
+
+```pseudo
+arrayListInstance.addAll(index, collection);
+```
+
+The contents of `collection` are inserted into the `arrayListInstance` at the specified `index`. The `collection` must be of the same [datatype](https://www.codecademy.com/resources/docs/java/data-types) as `arrayListInstance`.
```

Comment:
```suggestion
The contents of `collection` are inserted into the `arrayListInstance` at the specified `index`. The `collection` must be of the same [data type](https://www.codecademy.com/resources/docs/java/data-types) as `arrayListInstance`.
```

Referred to as "data type" in the [linked source](https://www.codecademy.com/resources/docs/java/data-types).